packages/containers-shared/src/utils.ts

```diff
@@ -250,3 +243,88 @@ export async function checkExposedPorts(
 export function generateContainerBuildId() {
 	return randomUUID().slice(0, 8);
 }
+
+type DockerContext = {
+	Current: boolean;
+	Description: string;
+	DockerEndpoint: string;
+	Error: string;
+	Name: string;
+};
+
+/**
+ * Run `docker context ls` to get the Docker socket from the currently active Docker context
+ */
+export function getDockerSocketFromContext(dockerPath: string) {
+	try {
+		const output = runDockerCmdWithOutput(dockerPath, [
+			"context",
+			"ls",
+			"--format",
+			"json",
+		]);
+
+		// Parse each line as a separate JSON object
+		const lines = output.trim().split("\n");
+		const contexts: DockerContext[] = lines.map((line) => JSON.parse(line));
+
+		// Find the current context
+		const currentContext = contexts.find((context) => context.Current === true);
+
+		if (
+			currentContext &&
+			currentContext.DockerEndpoint &&
+			typeof currentContext.DockerEndpoint === "string"
```

Comment:
The type says `DockerEndpoint: string;` so either this code is not needed or the type is wrong.

cf my comment about validation earlier